### PR TITLE
fix: Avoid cloning state changes.

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
Fixes: #2327

Adding a new function to finalize `TrieUpdate` that also returns state changes used for RPCs.

## Test plan:
- Unit tests / CI

### CODE REVIEW

File: runtime/runtime/src/lib.rs [hunk 1]
```diff
@@ -1074,10 +1074,9 @@ impl Runtime {
         )?;
 
         state_update.commit(StateChangeCause::UpdatedDelayedReceipts);
-        // TODO(#2327): Avoid cloning.
-        let state_changes = state_update.committed_updates_per_cause().clone();
```
Comment (frol) [inline runtime/runtime/src/lib.rs]: It seems it was the only place we used this method, so we should also remove its implementation.

File: core/store/src/trie/mod.rs [hunk 2]
```diff
@@ -25,6 +24,7 @@ use crate::trie::trie_storage::{
     TrieStorage,
 };
 use crate::{ColState, StorageError, Store, StoreUpdate};
+use near_primitives::utils::TrieKey;
```
Comment (frol) [inline core/store/src/trie/mod.rs]: If you get another commit, please, fix this use statement to be in the right group.

**Comment by frol** [comment]
@nearmax using `finalize_with_state_changes`  where `finalize` is enough would cast this optimization away since instead of this single clone we will end up with clones in every other place due to the fact of constructing `state_changes` vector. In this case, the invariant is aligned by the fact that both `finalize` and `finalize_with_state_changes` consume `self` and only the relevant data is extracted/returned back from the committed.

**Comment by evgenykuzyakov** [comment]
We can remove `finalize` because it's only used in tests. There are no other places, except for loading from genesis where we need to use `finalize`, so the production performance is not going to be affected.

**Comment by nearmax** [comment]
@frol If we were calling finalize outside tests then we would've been violating lifetime dependencies described here: https://github.com/nearprotocol/nearcore/issues/2348 :)

**Pull request merged**